plugins/org.eclipse.osee.framework.skynet.core/src/org/eclipse/osee/framework/skynet/core/artifact/RelationLoader.java

```diff
@@ -93,7 +108,20 @@ public static void loadRelationData(Long joinQueryId, Collection<Artifact> artif
       }
 
       for (Artifact artifact : artifacts) {
+         setTransactionRecord(artToTx, artifact);
          artifact.setLinksLoaded(true);
       }
    }
+
+   private static void setTransactionRecord(Map<ArtifactToken, TransactionToken> artToTx, Artifact artifact) {
+      TransactionToken txTok = artToTx.get(artifact);
+      if (txTok != null && txTok.isValid()) {
+         TransactionRecord txRec = TransactionManager.getTransactionRecord(txTok);
```

Comment:
Why is this needed? You've already got the TransactionToken above from the map(which is loaded from the db)

Reply:
TransactionRecord has all the data about transaction and is on every client Artifact.  Cheep to get and cache as transactions will never change and it keeps UI from having to load in all the places it needs the record.